awx/ui/src/screens/Setting/shared/SharedFields.js

```diff
@@ -229,6 +230,39 @@ EncryptedField.propTypes = {
   config: shape({}).isRequired,
 };
 
+const ExecutionEnvField = ({ name, config, isRequired = false }) => {
+  const validate = isRequired ? required(null) : null;
+  const [field, meta, helpers] = useField({ name, validate });
+  const isValid = !meta.error || !meta.touched;
+  return config ? (
+    <SettingGroup
+      defaultValue={config.default ?? ''}
+      fieldId={name}
+      helperTextInvalid={meta.error}
+      isRequired={isRequired}
+      label={config.label}
+      popoverContent={config.help_text}
+      validated={isValid ? 'default' : 'error'}
+      isDisabled={field.value === null}
+    >
+      <ExecutionEnvironmentLookup
+        onBlur={() => helpers.setTouched(true)}
+        value={field.value}
+        onChange={(value) => {
+          helpers.setValue(value);
+          helpers.setTouched(true);
```

Comment:
I think we should remove the setTouched here in favor of the one [above](https://github.com/ansible/awx/pull/11107/files#diff-edaa37419860b1661e289b954b771a9f5e10dfc50976c26d459f193ecc2221e4R249)